### DISCUSSION TITLE
Tonemap mode

### DISCUSSION THREAD

**Discussion by ozgurkeysan1** [google/model-viewer#2845] [Ideas]
Is it possible to make the tonemap mode in to an option for the model viewer? Should i make a feature request as an issue for this?

**Comment by ghost** [comment]
Can't change mode currently, setting exposure is available as an attribute however. Can you describe your intended use case, or how you'd like to use it?

**Reply by ozgurkeysan1** [reply]
We can say that our models are scanned and we want to be able to show them as close to the captured colors as possible

**Reply by elalish** [reply]
This is a very tricky thing in  practice! First, the scanned colors likely baked the effects of your lighting in,  so they are not material colors in the physically-based sense. If  you really just want the colors to come out  exactly as they went in you'll  want  to  use the KHR_materials_unlit extension. We do have an open bug on tone mapping being applied to unlit when  it shouldn't be. 

If you want to use lighting applied to your model, then the rendering is not supposed to match the base  color, since  it is affected by the applied lighting.

**Reply by ozgurkeysan1** [reply]
Yes we already have an open discussion here about forcing unlit behaviour on models without the extension metadata present so we're currently using a fallback by using an empty environment map and clearing all but base color texture and metal 0. So if forcing unlit is an easier change we can also do that.

**Reply by elalish** [reply]
Is it not possible to add the unlit extension metadata to your glTF files?

**Reply by ozgurkeysan1** [reply]
We're not in control of the conversion program and they don't support it but we've opened a discussion with them too about it and we also considered injecting the metadata too as a post processing step. At the same time it feels like a nice feature to be able to view the model as both lit and unlit at will?

**Reply by nighthoch** [reply]
> 
> 
> Is it not possible to add the unlit extension metadata to your glTF files?

im also curious about this issue, as were trying to work around the tonemapping affecting our models colors. do you have any advice for this process of "adding unlit extension metadata" ? im fairly novice and using blender to produce my 3d. just looking for some direction on how to go about that if you have any advice! thanks

**Reply by elalish** [reply]
Just  add 
```
"extensions": {
            "KHR_materials_unlit": {}
        }
```
to your material JSON, as described in the [extension](https://github.com/KhronosGroup/glTF/tree/main/extensions/2.0/Khronos/KHR_materials_unlit).
